Sources/SpeziCLAID/CLAIDRuntime.swift

```diff
@@ -0,0 +1,170 @@
+            
+            // Ensure pyCLAID is initialized and start it
+            guard let pyCLAIDLoader = self.pyCLAIDLoader else {
+                throw CLAIDError("Failed to start pyCLAID, object is null.")
+            }
+            _ = pyCLAIDLoader.attach_python_runtime()
+        } catch {
+            print("Error starting CLAID: \(error)")
+        }
+    }
+    
+    func setUpPythonDependencies() throws {
+        SwiftPython.startPythonInterpreter()
+        let sys = Python.import("sys")
+        // Ensure the script directory exists in the app bundle
+        if let scriptDirectory = Bundle.main.resourcePath {
+            
+            // Add necessary paths for Python dependencies.
+            // The claid_native framework contains the implementation of the Python runtime.
+            // Normally, this is available as pip package (pip install CLAID).
+            let packageDirectory = "\(scriptDirectory)/Frameworks/claid_native_xcframework.framework/"
+            sys.path.append(packageDirectory)
+            
+            // This folder contains the PyCLAIDLoader.py, so we have to add it to the Python path.
+            let bundleDir = "\(scriptDirectory)/SpeziCLAID_SpeziCLAID.bundle/Python"
+            sys.path.append(bundleDir)
+            
+            // Add the current Bundle as path as well, enabling us to import Python files from the Resources.
+            sys.path.append(scriptDirectory)
+            
+            self.pyCLAIDLoader = Python.import("PyCLAIDLoader")
+            
+            // Ensure pyCLAID is successfully imported
+            guard let pyCLAIDLoader = self.pyCLAIDLoader else {
+                throw CLAIDError("Failed to start pyCLAID, failed to import PyCLAIDLoader, object is null.")
+            }
+            
+            // Import and add each registered Python module
+            try CLAIDRuntime.moduleRegistry.pythonModuleDetails.forEach { moduleInfo in
+                let result = pyCLAIDLoader.import_and_add_module(
+                    scriptDirectory,
+                    moduleInfo.pythonFilePath,
+                    moduleInfo.pythonModuleName
+                )
+                
+                if Bool(result) == false {
+                    throw CLAIDError("Failed to import PythonModule \(moduleInfo.pythonModuleName) from file \(moduleInfo.pythonFilePath)." +
+                                     "File not found or does not contain class '\(moduleInfo.pythonModuleName)'. Check files under \(scriptDirectory).")
+                }
+            }
+        }
+        
+    }
+    
+  
+    
+    /// Initializes the Python interpreter and starts CLAID
+    /// - Throws: CLAIDError if initialization fails
+    public func start() throws {
+        try setUpPythonDependencies()
+        try writeOutCLAIDConfig(config: self.config)
+        
+        // Start asynchronous initialization.
+        Task {
+            await self.startCLAIDFramework()
+        }
+    }
```

Comment:
Wondering if it makes sense to make the complete function asynchronous to allow one to observe the overall startup behavior? We will have to handle the overall mechanisms in the configure function, but it would allow someone outside of Spezi to still enable all this behavior if they want to.